homeworks/06-advanced-redux/src/App.tsx

```diff
@@ -12,21 +12,31 @@ import {
   TotalPrice,
 } from './components';
 
-function App() {
-  const {
-    addItemToCart,
-    cart,
-    menu,
-    removeItemFromCart,
-    totalPrice,
-  } = useApp();
+import { shop } from './redux/store';
+import { addToCart, removeFromCart, selectItem } from './redux/actions/shop';
+import { loadMenuAsync } from './redux/actions/menu';
+
+import { Pizza, ShopState } from './types';
+
+interface AppProps {
+  state: ShopState
+}
+
+function App(props: AppProps) {
+  const { menu, cart } = props.state;
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(loadMenuAsync());
+  }, []);
 
   const handleRemoveItemFromCart = useCallback((_id: string) => {
-    removeItemFromCart(_id);
+    shop.dispatch(removeFromCart(_id));
```

Comment:
why not `useDispatch` here?

Reply:
Updated to use useDispatch, see https://github.com/vbarbasins/evo-ts-bootcamp/pull/7/commits/40000cf19231ba1b304ed01ab3879930296ac2a9